benchmarks/bench_ml.py

```diff
@@ -0,0 +1,422 @@
+                'max_depth': (Real(1.0, 4.0), pow2intmap),
+                'min_samples_split': (Real(1.0, 8.0), pow2intmap),
+            },
+            MODEL_BACKEND: DecisionTreeClassifier,
+        },
+    }
+
+    # every dataset should have fields:
+    # OUTPUT_TYPE:  OUTPUT_CONTINUOUS / OUTPUT_CATEGORY
+    # DATASET_LOADER: callable that loads data into matrix X of inputs and vector y of outputs
+    datasets={
+        "Boston":{
+            OUTPUT_TYPE: OUTPUT_CONTINUOUS,
+            DATASET_LOADER: lambda : load_data_target(skd.load_boston())
+        },
+        "Housing":{
+            OUTPUT_TYPE: OUTPUT_CONTINUOUS,
+            DATASET_LOADER: lambda : load_data_target(skd.fetch_california_housing())
+        },
+        #
+        #   classification datasets go here
+        #
+        "Leukemia":{
+            OUTPUT_TYPE: OUTPUT_CATEGORY,
+            DATASET_LOADER: lambda : load_data_target(skd.fetch_mldata('leukemia'))
+        },
+        "Climate Model Crashes":{
+            OUTPUT_TYPE: OUTPUT_CATEGORY,
+            DATASET_LOADER: lambda : load_data_target(skd.fetch_mldata('climate-model-simulation-crashes', target_name=-1))
+        },
+    }
+
+    @staticmethod
+    def supports(model, dataset):
+        """
+        Used to determine if model can be trained on particular dataset.
+        :param model: string, name of model class
+        :param dataset: string, name of dataset
+        :return: bool
+        """
+
+        if not dataset in TesterClass.datasets:
+            return False, "unknown dataset"
+
+        selected_dataset = TesterClass.datasets[dataset]
+        if not model in TesterClass.models:
+            return False, "unknown model"
+
+        selected_model = TesterClass.models[model]
+        if not selected_dataset[OUTPUT_TYPE] == selected_model[OUTPUT_TYPE]:
+            return False, "model cannot be applied to the dataset"
+
+        return True, None
+
+    def __init__(self, model, dataset):
+
+        supports, reason_if_not = TesterClass.supports(model, dataset)
+        if not supports:
+            raise BaseException(reason_if_not)
+
+        X, Y = self.datasets[dataset][DATASET_LOADER]()
+
+        self.dataset = split_normalize(X, Y)
+        self.model_class = TesterClass.models[model]
+
+    def evaluate(self, point):
+        """
+        Evaluates model on loaded dataset for particular setting of hyperparameters.
+        :param point: dict, pairs of parameter names and corresponding values
+        :return: score (more is better!) for some specific point
+        """
+
+        X, Y, Xv, Yv = self.dataset
+        cls = self.model_class[MODEL_BACKEND]
+
+        # apply transformation to model parameters, for example exp transformation
+        point_mapped = {}
+
+        for param in point.keys():
+            v = point[param]
+            v = self.model_class[MODEL_PARAMETERS][param][1](v)
+            point_mapped[param] = v
+
+        model = cls(**point_mapped)
```

Comment:
The above block of code applies transformation to the values of the point being evaluated, so for example to the value of C = -1.0 the exponent is applied so that it becomes 0.1 (there are some other transformations like rounding and integer exponentiation). This is what the loop above the cls(**point_mapped) is for. With how things are right now I can just write `self.model_class[MODEL_BACKEND].(**point_mapped)`, where the point_mapped contains transformed values.